client/src/components/EmailFrontend/EmailFrontend.js

```diff
@@ -162,20 +164,22 @@ class EmailFrontend extends React.Component {
 
     const { include_privacy } = this.props;
 
-    const user_fields = _.omitBy(
-      template,
-      ({ form_type }, key) => key === "meta" || !form_type
-    );
+    const user_fields = _.omitBy(template, ({ form_type }, key) => {
+      key === "meta" || !form_type;
+    });
 
-    const all_required_user_fields_are_filled = _.chain(user_fields)
-      .omitBy((field) => !field.required)
-      .keys()
-      .every(
-        (required_field_key) =>
-          !_.isUndefined(completed_template[required_field_key]) &&
-          !_.isEmpty(completed_template[required_field_key])
-      )
-      .value();
+    const all_required_user_fields_are_filled =
+      // specifically for storybook, will not return true
+      this.props.all_required_user_fields_are_filled ||
```

Comment:
Delete, not necessary if above is fixed. Besides which, if a component really needs to be modified for the sake of storybook testing, then the component should probably be refactored in general but, I think, it should never be left containing specific artifacts related to the story. The stories need to make the components easier to maintain, but if they add weird special props to the interface then they're probably only making things worse (absolutely worst case, using this prop in the actual client would need to some how throw an error, based of environment variables or something),

Reply:
Done! Thanks for the advice.